urllib3/sessionmanager.py

```diff
@@ -0,0 +1,87 @@
+from .util.retry import Retry
+from .request import RequestMethods, Request
+from .poolmanager import PoolManager
+from .util.context import Context
+
+
+class SessionManager(RequestMethods):
+    """
+    Allows arbitrary requests while maintaining session context across
+    those requests. Currently, that context consists of automatic
+    cookie storage and retrieval.
+
+    :param headers:
+        Headers to include with all requests, unless other headers are
+        given explcitly
+
+    :param cookie_jar:
+        A predefined :class:`CookieJar` object to use in the session;
+        if not provided, a new one will be created.
+
+    :param \**pm_kwargs:
+        Additional paramaters to be passed to the encapsulated
+        :class:`.PoolManager` or :class:`.ProxyManager` instance.
+    """
+    def __init__(self, manager, cookie_jar=None, headers=None):
+        super(SessionManager, self).__init__(headers=headers)
+        self.manager = manager
+        self.context = Context(cookie_jar)
```

Comment:
I'll add documentation around it; I didn't update the docs after switching SessionManager to receive a PoolManager in init.